File: 83. Remove Duplicates from Sorted List/removeDuplicates.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        //Make a pointer pointing to the head
        ListNode current_node = head;
        
        //While the current_node is not null, traverse and delete any duplicates
        while(current_node != null && current_node.next != null){
            if (current_node.val == current_node.next.val){
                current_node.next = current_node.next.next;
            }else{
            //Make sure you have an else statement in here otherwise it won't work
            current_node = current_node.next;
            }
        }
        return head;
    }
}

File: 83. Remove Duplicates from Sorted List/removeDuplicates.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* deleteDuplicates(struct ListNode* head){
    //Make a pointer to the head so you can traverse
    struct ListNode* current_node = head;
    //Edge case: linked list has no elements
    if (current_node == NULL) return head;
    
    while (current_node->next != NULL){
        if (current_node->val == current_node->next->val){
            current_node->next = current_node->next->next;
        }else{
            current_node = current_node->next;
        }
    }
    return head;  
    }
